### PR TITLE
created project overview page

### PR DESCRIPTION
**Pull request by UmaySoyluer**
Please have a look 🌱

### CODE REVIEW

**Comment by TheRealChio** [comment]
added env variable to vercel

File: components/ProjectList.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    return <Error message={error.message} />;
+  }
+
+  return (
+    <>
+      {projects?.length
+        ? projects.map((project) => (
+            <Project key={project.id} project={project} />
+          ))
+        : "No projects found."}
```
Comment (julialoeschel) [inline components/ProjectList.js:21]: you could also wrap this in a semantic HTML element.

File: lib/fetcher.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default async function fetcher(url) {
+  const res = await fetch(url);
```
Comment (julialoeschel) [inline lib/fetcher.js:2]: please stick to good coding practice and write full words (always).

File: lib/fetcher.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  if (!res.ok) {
+    const error = new Error("An error occurred while fetching the data.");
+    // Attach extra info to the error object.
+    error.info = await res.json();
+    error.message = res.message;
```
Comment (julialoeschel) [inline lib/fetcher.js:8]: the response object og the fetch method does not have a message property. [What you are looking for is `statusText`](https://developer.mozilla.org/en-US/docs/Web/API/Response)

**Review by julialoeschel — CHANGES_REQUESTED** [review]
great start!

I do have some things to remind you of: 
- No empty files. please delete every file that you dont need to fulfill the userstory. 
- your deployment is failing, please make sure to always provide this.
- you are missing the index.js file (this might be the reason for the failing) and your app does not have a homepage, and therefore is nor usable for the user. 

make sure you always have a working app, even with your first PR :)

File: components/ProjectList.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import Project from "./Project";
+import Error from "@/components/Error";
+import useSWR from "swr";
+
+export default function ProjectList() {
+  const { data: projects, isLoading, error } = useSWR("/api/projects");
+
+  if (isLoading) {
+    return <h1>Loading ... </h1>;
```
Comment (julialoeschel) [inline components/ProjectList.js:9]: please only use h1 once overall your page (in the header)

**Review by julialoeschel — CHANGES_REQUESTED** [review]
great, good work!

one more thing: h1 only should appear once overall your app (the most important heading). Its a very strong semantic element, that is relevant for e.g. SEO, screenreader and other asssistive devices. 

make sure you dont mess up your app and change to other elements like `<p>`or `h2`or even `<span>`if you need to cover text.

File: components/Error.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default function Error({ message }) {
+  return <h1>Error! {message}</h1>;
```
Comment (julialoeschel) [inline components/Error.js:2]: please only use h1 once overall your page (in the header)

File: components/Heading.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export default function Heading({ children }) {
+  return <h1>{children}</h1>;
```
Comment (julialoeschel) [inline components/Heading.js:2]: please only use h1 once overall your page (in the header)

File: components/Project.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export default function Project({ project }) {
+  return (
+    <li>
+      <h1>{project.title}</h1>
```
Comment (julialoeschel) [inline components/Project.js:4]: please only use h1 once overall your page (in the header)

File: components/ProjectList.js (hunk 4 above)
Comment (UmaySoyluer) [reply components/ProjectList.js:9]: thanks Julia. we are on it now

**Pull request merged**